### ISSUE TITLE
FP16 run stagnates at "OVERFLOW! Skipping step. Attempted loss scale: 1"

### ISSUE THREAD

**Issue by theconsummate** [NVIDIA/tacotron2#112]
I am using pytorch 1.0 and python 3.6 with a single Tesla V100 gpu. I tried to train with fp16 support using the latest code on the master branch.

```
python train.py --output_directory=outdir --log_directory=logdir --hparams=fp16_run=True
```

The attempted loss scale decays to 1 and the fp16_optimizer still keeps detecting overflow.

Logs:

FP16 Run: True
Dynamic Loss Scaling: True
Distributed Run: False
cuDNN Enabled: True
cuDNN Benchmark: False
...

FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([80])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([80])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([80])
Epoch: 0
OVERFLOW! Skipping step. Attempted loss scale: 4294967296
OVERFLOW! Skipping step. Attempted loss scale: 2147483648.0
OVERFLOW! Skipping step. Attempted loss scale: 1073741824.0
OVERFLOW! Skipping step. Attempted loss scale: 536870912.0
...
OVERFLOW! Skipping step. Attempted loss scale: 262144.0
/home/ubuntu/tts/tacotron2/fp16_optimizer.py:173: UserWarning: torch.nn.utils.clip_grad_norm is now deprecated in favor of torch.nn.utils.clip_grad_norm_.
  return torch.nn.utils.clip_grad_norm(fp32_params, clip)
Train loss 15 47.531357 Grad Norm 11.616506 2.96s/it
Train loss 16 35.477085 Grad Norm 19.570642 2.86s/it
Train loss 17 17.105583 Grad Norm 16.489749 2.89s/it
Train loss 18 10.484207 Grad Norm 14.732778 2.99s/it
OVERFLOW! Skipping step. Attempted loss scale: 131072.0
OVERFLOW! Skipping step. Attempted loss scale: 65536.0
OVERFLOW! Skipping step. Attempted loss scale: 32768.0
...
OVERFLOW! Skipping step. Attempted loss scale: 512.0
OVERFLOW! Skipping step. Attempted loss scale: 256.0
OVERFLOW! Skipping step. Attempted loss scale: 128.0
OVERFLOW! Skipping step. Attempted loss scale: 64.0
OVERFLOW! Skipping step. Attempted loss scale: 32.0
OVERFLOW! Skipping step. Attempted loss scale: 16.0
OVERFLOW! Skipping step. Attempted loss scale: 8.0
OVERFLOW! Skipping step. Attempted loss scale: 4.0
OVERFLOW! Skipping step. Attempted loss scale: 2.0
OVERFLOW! Skipping step. Attempted loss scale: 1.0
OVERFLOW! Skipping step. Attempted loss scale: 1
OVERFLOW! Skipping step. Attempted loss scale: 1
OVERFLOW! Skipping step. Attempted loss scale: 1
OVERFLOW! Skipping step. Attempted loss scale: 1
...

this goes on for a couple of epochs after which I stopped

**Comment by FredericGodin** [comment]
Same issue here with Nvidia RTX 2080 Ti card...
Any solution yet?

**Comment by rafaelvalle** [comment]
Try this pull request: https://github.com/NVIDIA/tacotron2/pull/137

**Comment by prajwalkr** [comment]
> Try this pull request: #137

It did not work, the issue still persists for me as well.

**Comment by ddemilla** [comment]
Same issue on an RTX 2070

**Comment by rafaelvalle** [comment]
We are validating a fix to this. For those who do not want to wait, remove or comment these two lines that apply dropout to the lstm cell states:
https://github.com/NVIDIA/tacotron2/blob/master/model.py#L358
https://github.com/NVIDIA/tacotron2/blob/master/model.py#L375

It should also work when using the pre-trained model to train other models.

**Comment by rafaelvalle** [comment]
This issued should be fixed.
